src/FilenameParsing/HashFileIDHelper.php

```diff
@@ -65,6 +66,17 @@ public function lookForVariantIn(ParsedFileID $parsedFileID)
         return  $folder . $this->truncate($parsedFileID->getHash());
     }
 
+    public function getVariantGlob(string $folder, ParsedFileID $parsedFileID): string
```

Comment:
Move this to AbstractFileIDHelper where you doing something like this:

```php
return pathinfo($path, PATHINFO_FILENAME) . FileIDHelper::VARIANT_SEPARATOR . '*';
```

Where `PATHINFO_FILENAME` is the filename minus the path and extensions e.g. path/to/my.jpg = "my"

Do the same for NaturalFileIDHelper.  For both of these methods we're just getting $fileWithoutExtension and then suffixing '__*'

Reply:
The abstract helper doesn't (and shouldn't) implement `GlobbableFileIDHelper`.

Someone might have subclassed `AbstractFileIDHelper` to implement their own logic with their own naming convention for variants. If we suddenly pop this into the abstract helper, we're likely to be providing an incorrect implementation of this method for their helper, which will break the `lookForVariantIn` functionality when using their helper.

Reply:
I'm also not sure what you're trying to achieve with that suggestion? Is there some duplication of code you're trying to avoid, or is it for some other purpose?

Reply:
Just to remove code duplication.  Originally I was thinking if we could get rid of the new interface, though that's not really possible.

I guess just update the methods in HashFileIDHelper / NaturalFileIDHelper directly then, could use a trait for this.

Reply:
I'm not adding a whole trait just for a single line of code - I assume that's not what you actually intended, but that's what it _seems_ like you're implying should be done.

I *think* what you actually meant is that `pathinfo($path, PATHINFO_FILENAME)` should be used instead of the regex replacements. I'll see if that's doable since that seems sensible.

Reply:
Ah right OK. In that can just update the names of the variables those methods, or add comments, to reflect the fact that "$file" is not exclusively "$file" with no "$folder"

Also, why are we removing any of the folder, wouldn't the glob be better with all of the folder there?

Reply:
> why are we removing any of the folder, wouldn't the glob be better with all of the folder there?

If you glob for `dir1/dir2/file_*.jpg` when you're already inside `dir1`, you won't find anything. Because dir1 isn't inside dir1.

Reply:
> In that can just update the names of the variables those methods

Done